include/pybind11/pybind11.h

```diff
@@ -143,8 +143,11 @@ class cpp_function : public function {
             /* Override policy for rvalues -- usually to enforce rvp::move on an rvalue */
             const auto policy = detail::return_value_policy_override<Return>::policy(call.func.policy);
 
+            /* Function scope guard -- defaults to the compile-to-nothing `void_type` */
```

Comment:
Scratch that, I see now: it's going to be `call_guard<>::type`, which is the `void_type`.